src/controllers/VueController.php

```diff
@@ -25,9 +22,15 @@ public function __construct($id, $module, $config = [])
     {
         parent::__construct($id, $module, $config);
 
-        $this->protocol   = (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' || $_SERVER['SERVER_PORT'] == 443) ? "https://" : "http://";
-        $this->start_date = $_POST['range_start'] ?? null;
-        $this->end_date   = $_POST['range_end'] ?? null;
+        $today          = new \DateTime(date('Y-m-d'));
+        $weekAgo        = new \DateTime(date('Y-m-d'));
+        $weekAgo        = $weekAgo->modify('-7 day')->format('Y-m-d 00:00:00');
+        $rangeStart     = \Craft::$app->request->getBodyParam('range_start');
```

Comment:
You could also add `use Craft;` to import the Craft and not need to add `\` everywhere.